Scripts/Engine/CustomBar.lua

```diff
@@ -0,0 +1,848 @@
+						"BLANK", TEN.Vec2(0,0), {}, 0, TEN.Color(0,0,0), true, alphaBlendSpeed, blink, blinkLimit)
+		
+		-- LevelVars.Engine.CustomBars.bars[dataName].oldValue			= maxValue
+	end
+
+	
+	LevelVars.Engine.CustomBars.bars[dataName].getActionType	= getActionType
+	LevelVars.Engine.CustomBars.bars[dataName].showBar			= showBar
+	LevelVars.Engine.CustomBars.bars[dataName].visible			= true
+	LevelVars.Engine.CustomBars.bars[dataName].targetAlpha		= 255
+
+	
+end
+
+---
+-- Creates a custom progress bar for an enemy with extensive configuration options. Ensure this function is called before Lara aims at the enemy if using generic enemy HP bars.
+-- @tparam string barName Unique name for the bar.
+-- @tparam Objects.ObjID objectIDbg Object ID for the bar's background sprite.
+-- @tparam number spriteIDbg SpriteID from the specified object for the bar's background.
+-- @tparam Color colorbg Color of bar's background.
+-- @tparam Vec2 posBG X,Y position of the bar's background in screen percent (0-100).
+-- @tparam number rotBG rotation of the bar's background. sprite (0-360).
+-- @tparam Vec2 scaleBG X,Y Scaling factor for the bar's background sprite.
+-- @tparam View.AlignMode alignModebg Alignment for the bar's background.
+-- @tparam View.ScaleMode scaleModebg Scaling for the bar's background.
+-- @tparam Effects.BlendID blendModebg Blending modes for the bar's background.
+-- @tparam Objects.ObjID objectIDbar Object ID for the bar sprite.
+-- @tparam number spriteIDbar SpriteID from the specified object for the bar.
+-- @tparam Color colorbar Color of the bar.
+-- @tparam Vec2 posBar X,Y position of the bar in screen percent (0-100).
+-- @tparam number rot rotation of the bar's sprite (0-360).
+-- @tparam Vec2 scaleBar X,Y Scaling factor for the bar's sprite.
+-- @tparam View.AlignMode alignMode Alignment for the bar.
+-- @tparam View.ScaleMode scaleMode Scaling for the bar.
+-- @tparam Effects.BlendID blendMode Blending modes for the bar.
+-- @tparam string text Text to display for the enemy.
+-- @tparam Vec2 textPos X,Y position of the text.
+-- @tparam Strings.DisplayStringOption textOptions alignment and effects for the text. Default: None. Please note text is automatically aligned to the LEFT
+-- @tparam number textScale Scale factor for the text.
+-- @tparam Color textColor Color of the text.
+-- @tparam bool hideText Whether to hide the text.
+-- @tparam number alphaBlendSpeed Speed of alpha blending for bar visibility (0-255).
+-- @tparam string object Enemy name set in Editor for which to create HP for.
+-- @tparam bool showBar Option to always show the bar whether the enemy is current target or not. Useful for boss health bars.
+-- @tparam bool blink Whether the bar blinks.
+-- @tparam number blinkLimit %Limit below which bar starts blinking (0-1).
+--
+-- @treturn CustomBar The custombar in its hidden state
+CustomBar.CreateEnemyHpBar = function (barName,

```

Comment:
Same as above. Try to use setup table to collect all parameters.

Reply:
Done